File: DragonzLand/DragonzLandBot.vb
Imports System.Net.Http
Imports System.Text
Imports System.Text.Json
Imports System.Threading

Public Class DragonzLandBot

    Private ReadOnly PubQuery As DragonzLandQuery
    Private ReadOnly AccessToken As String
    Public ReadOnly HasError As Boolean
    Public ReadOnly ErrorMessage As String

    Public Sub New(Query As DragonzLandQuery)
        PubQuery = Query
        Dim GetToken = DragonzLandGetToken().Result
        If GetToken IsNot Nothing Then
            AccessToken = GetToken.AccessToken
            HasError = False
            ErrorMessage = ""
        Else
            HasError = True
            ErrorMessage = "get token failed"
        End If
    End Sub

    Private Async Function DragonzLandGetToken() As Task(Of DragonzLandAuthResponse)
        Dim DLAPI As New DragonzLandApi(0, PubQuery.Auth, PubQuery.Index)
        Dim request As New DragonzLandAuthRequest With {.InitData = PubQuery.Auth}
        Dim serializedRequest = JsonSerializer.Serialize(request)
        Dim serializedRequestContent = New StringContent(serializedRequest, Encoding.UTF8, "application/json")
        Dim httpResponse = Await DLAPI.DLAPIPost("https://bot.dragonz.land/api/auth/telegram", serializedRequestContent)
        If httpResponse IsNot Nothing Then
            If httpResponse.IsSuccessStatusCode Then
                Dim responseStream = Await httpResponse.Content.ReadAsStreamAsync()
                Dim responseJson = Await JsonSerializer.DeserializeAsync(Of DragonzLandAuthResponse)(responseStream)
                Return responseJson
            Else
                Return Nothing
            End If
        Else
            Return Nothing
        End If
    End Function

    Public Async Function DragonzLandUserDetail() As Task(Of DragonzLandMeResponse)
        Dim rDLAPI As New DragonzLandApi(1, AccessToken, PubQuery.Index)
        Dim rewardResponse = Await rDLAPI.DLAPIGet("https://bot.dragonz.land/api/tasks/welcome-reward")
        Thread.Sleep(3000)
        Dim DLAPI As New DragonzLandApi(1, AccessToken, PubQuery.Index)
        Dim httpResponse = Await DLAPI.DLAPIGet("https://bot.dragonz.land/api/me")
        If httpResponse IsNot Nothing Then
            If httpResponse.IsSuccessStatusCode Then
                Dim responseStream = Await httpResponse.Content.ReadAsStreamAsync()
                Dim responseJson = Await JsonSerializer.DeserializeAsync(Of DragonzLandMeResponse)(responseStream)
                Return responseJson
            Else
                Return Nothing
            End If
        Else
            Return Nothing
        End If
    End Function

    Public Async Function DragonzLandTasks() As Task(Of List(Of DragonzLandTaskResponse))
        Dim DLAPI As New DragonzLandApi(1, AccessToken, PubQuery.Index)
        Dim httpResponse = Await DLAPI.DLAPIGet("https://bot.dragonz.land/api/tasks")
        If httpResponse IsNot Nothing Then
            If httpResponse.IsSuccessStatusCode Then
                Dim responseStream = Await httpResponse.Content.ReadAsStreamAsync()
                Dim responseJson = Await JsonSerializer.DeserializeAsync(Of List(Of DragonzLandTaskResponse))(responseStream)
                Return responseJson
            Else
                Return Nothing
            End If
        Else
            Return Nothing
        End If
    End Function

    Public Async Function DragonzLandTasksVerify(taskId As String) As Task(Of Boolean)
        Dim DLAPI As New DragonzLandApi(1, AccessToken, PubQuery.Index)
        Dim request As New DragonzLandTaskDoneRequest With {.TaskId = taskId}
        Dim serializedRequest = JsonSerializer.Serialize(request)
        Dim serializedRequestContent = New StringContent(serializedRequest, Encoding.UTF8, "application/json")
        Dim httpResponse = Await DLAPI.DLAPIPost("https://bot.dragonz.land/api/me/tasks/verify", serializedRequestContent)
        If httpResponse IsNot Nothing Then
            If httpResponse.IsSuccessStatusCode Then
                Return True
            Else
                Return False
            End If
        Else
            Return False
        End If
    End Function

    Public Async Function DragonzLandFeed(feedCount As Integer) As Task(Of Boolean)
        Dim DLAPI As New DragonzLandApi(1, AccessToken, PubQuery.Index)
        Dim request As New DragonzLandFeedRequest With {.FeedCount = feedCount}
        Dim serializedRequest = JsonSerializer.Serialize(request)
        Dim serializedRequestContent = New StringContent(serializedRequest, Encoding.UTF8, "application/json")
        Dim httpResponse = Await DLAPI.DLAPIPost("https://bot.dragonz.land/api/me/feed", serializedRequestContent)
        If httpResponse IsNot Nothing Then
            If httpResponse.IsSuccessStatusCode Then
                Return True
            Else
                Return False
            End If
        Else
            Return False
        End If
    End Function

    Public Async Function DragonzLandBuyBoost(boostId As String) As Task(Of Boolean)
        Dim DLAPI As New DragonzLandApi(1, AccessToken, PubQuery.Index)
        Dim request As New DragonzLandBoostBuyRequest With {.BoostId = boostId}
        Dim serializedRequest = JsonSerializer.Serialize(request)
        Dim serializedRequestContent = New StringContent(serializedRequest, Encoding.UTF8, "application/json")
        Dim httpResponse = Await DLAPI.DLAPIPost("https://bot.dragonz.land/api/me/boosts/buy", serializedRequestContent)
        If httpResponse IsNot Nothing Then
            If httpResponse.IsSuccessStatusCode Then
                Return True
            Else
                Return False
            End If
        Else
            Return False
        End If
    End Function

End Class

File: DragonzLand/DragonzLandBot.cs
using System.Collections.Generic;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading;
using System.Threading.Tasks;

namespace DragonzLandBot
{

    public class DragonzLandBot
    {

        private readonly DragonzLandQuery PubQuery;
        private readonly string AccessToken;
        public readonly bool HasError;
        public readonly string ErrorMessage;

        public DragonzLandBot(DragonzLandQuery Query)
        {
            PubQuery = Query;
            var GetToken = DragonzLandGetToken().Result;
            if (GetToken is not null)
            {
                AccessToken = GetToken.AccessToken;
                HasError = false;
                ErrorMessage = "";
            }
            else
            {
                HasError = true;
                ErrorMessage = "get token failed";
            }
        }

        private async Task<DragonzLandAuthResponse> DragonzLandGetToken()
        {
            var DLAPI = new DragonzLandApi(0, PubQuery.Auth, PubQuery.Index);
            var request = new DragonzLandAuthRequest() { InitData = PubQuery.Auth };
            string serializedRequest = JsonSerializer.Serialize(request);
            var serializedRequestContent = new StringContent(serializedRequest, Encoding.UTF8, "application/json");
            var httpResponse = await DLAPI.DLAPIPost("https://bot.dragonz.land/api/auth/telegram", serializedRequestContent);
            if (httpResponse is not null)
            {
                if (httpResponse.IsSuccessStatusCode)
                {
                    var responseStream = await httpResponse.Content.ReadAsStreamAsync();
                    var responseJson = await JsonSerializer.DeserializeAsync<DragonzLandAuthResponse>(responseStream);
                    return responseJson;
                }
                else
                {
                    return null;
                }
            }
            else
            {
                return null;
            }
        }

        public async Task<DragonzLandMeResponse> DragonzLandUserDetail()
        {
            var rDLAPI = new DragonzLandApi(1, AccessToken, PubQuery.Index);
            var rewardResponse = await rDLAPI.DLAPIGet("https://bot.dragonz.land/api/tasks/welcome-reward");
            Thread.Sleep(3000);
            var DLAPI = new DragonzLandApi(1, AccessToken, PubQuery.Index);
            var httpResponse = await DLAPI.DLAPIGet("https://bot.dragonz.land/api/me");
            if (httpResponse is not null)
            {
                if (httpResponse.IsSuccessStatusCode)
                {
                    var responseStream = await httpResponse.Content.ReadAsStreamAsync();
                    var responseJson = await JsonSerializer.DeserializeAsync<DragonzLandMeResponse>(responseStream);
                    return responseJson;
                }
                else
                {
                    return null;
                }
            }
            else
            {
                return null;
            }
        }

        public async Task<List<DragonzLandTaskResponse>> DragonzLandTasks()
        {
            var DLAPI = new DragonzLandApi(1, AccessToken, PubQuery.Index);
            var httpResponse = await DLAPI.DLAPIGet("https://bot.dragonz.land/api/tasks");
            if (httpResponse is not null)
            {
                if (httpResponse.IsSuccessStatusCode)
                {
                    var responseStream = await httpResponse.Content.ReadAsStreamAsync();
                    var responseJson = await JsonSerializer.DeserializeAsync<List<DragonzLandTaskResponse>>(responseStream);
                    return responseJson;
                }
                else
                {
                    return null;
                }
            }
            else
            {
                return null;
            }
        }

        public async Task<bool> DragonzLandTasksVerify(string taskId)
        {
            var DLAPI = new DragonzLandApi(1, AccessToken, PubQuery.Index);
            var request = new DragonzLandTaskDoneRequest() { TaskId = taskId };
            string serializedRequest = JsonSerializer.Serialize(request);
            var serializedRequestContent = new StringContent(serializedRequest, Encoding.UTF8, "application/json");
            var httpResponse = await DLAPI.DLAPIPost("https://bot.dragonz.land/api/me/tasks/verify", serializedRequestContent);
            if (httpResponse is not null)
            {
                if (httpResponse.IsSuccessStatusCode)
                {
                    return true;
                }
                else
                {
                    return false;
                }
            }
            else
            {
                return false;
            }
        }

        public async Task<bool> DragonzLandFeed(int feedCount)
        {
            var DLAPI = new DragonzLandApi(1, AccessToken, PubQuery.Index);
            var request = new DragonzLandFeedRequest() { FeedCount = feedCount };
            string serializedRequest = JsonSerializer.Serialize(request);
            var serializedRequestContent = new StringContent(serializedRequest, Encoding.UTF8, "application/json");
            var httpResponse = await DLAPI.DLAPIPost("https://bot.dragonz.land/api/me/feed", serializedRequestContent);
            if (httpResponse is not null)
            {
                if (httpResponse.IsSuccessStatusCode)
                {
                    return true;
                }
                else
                {
                    return false;
                }
            }
            else
            {
                return false;
            }
        }

        public async Task<bool> DragonzLandBuyBoost(string boostId)
        {
            var DLAPI = new DragonzLandApi(1, AccessToken, PubQuery.Index);
            var request = new DragonzLandBoostBuyRequest() { BoostId = boostId };
            string serializedRequest = JsonSerializer.Serialize(request);
            var serializedRequestContent = new StringContent(serializedRequest, Encoding.UTF8, "application/json");
            var httpResponse = await DLAPI.DLAPIPost("https://bot.dragonz.land/api/me/boosts/buy", serializedRequestContent);
            if (httpResponse is not null)
            {
                if (httpResponse.IsSuccessStatusCode)
                {
                    return true;
                }
                else
                {
                    return false;
                }
            }
            else
            {
                return false;
            }
        }

    }
}